src/xutils.cpp

```diff
@@ -50,37 +50,6 @@ namespace xcpp
         exit(0);
     }
 
-    bool should_print_version(int argc, char* argv[])
-    {
-        for (int i = 0; i < argc; ++i)
-        {
-            if (std::string(argv[i]) == "--version")
-            {
-                return true;
-            }
-        }
-        return false;
-    }
-
-    std::string extract_filename(int &argc, char* argv[])
-    {
-        std::string res = "";
-        for (int i = 0; i < argc; ++i)
-        {
-            if ((std::string(argv[i]) == "-f") && (i + 1 < argc))
-            {
-                res = argv[i + 1];
-                for (int j = i; j < argc - 2; ++j)
-                {
-                    argv[j] = argv[j + 2];
-                }
-                argc -= 2;
-                break;
-            }
-        }
-        return res;
-    }
-
```

Comment:
I'd keep the tests as upstream testing was a bit lacking last time I saw.

Reply:
Ahhh no worries I made sure to add a test here (https://github.com/anutosh491/xeus/blob/13aea3049c99df566ae54061432b57b100747e91/test/test_unit_kernel.cpp#L56-L65)